src/Mutex.php

```diff
@@ -89,12 +85,14 @@ class Mutex extends \yii\mutex\Mutex
      * This method will initialize the [[redis]] property to make sure it refers to a valid redis connection.
      * @throws InvalidConfigException if [[redis]] is invalid.
      */
-    public function init()
+    public function __construct(Connection $redis, $autoRelease = true)
     {
-        parent::init();
-        $this->redis = Instance::ensure($this->redis, Connection::className());
+        parent::__construct($autoRelease);
+        $this->redis = $redis;
+
+        // FIXME: needs to be set elsewhere
         if ($this->keyPrefix === null) {
-            $this->keyPrefix = substr(md5(Yii::$app->id), 0, 5);
+            $this->keyPrefix = substr(md5(Yii::getApp()->id), 0, 5);
```

Comment:
Waiting to hear on https://github.com/yiisoft/mutex/issues/5 to see how to properly handle setting `keyPrefix`

Reply:
Remove `=== null` condition.